isort_per_file/__init__.py

```diff
@@ -0,0 +1,37 @@
+"""Precommit hook that runs flake8 on one file at a time.
+
+This allows isort to properly find the .isort.cfg for each subpackage.
+
+Clone of the flake8 handling system, but without the manual .cfg detection
+
+Extremely badly written; does not expect any arguments other than files
+to run isort on. Could likely run on all files from a given subpackage
+ but this is simpler.
+"""
+
+import os.path
+import pathlib
+import subprocess
+import sys
+
+
+
+def main():
+    my_flake8 = os.path.join(os.path.dirname(sys.argv[0]), 'flake8')
```

Comment:
You want this to be `my_isort`

Reply:
Can I test this before merging?

Reply:
Yes! I think you can edit your `.pre-commit-config.yaml` in `deus_lex` to invoke this branch, though you do have to then do a bit of extra work to tell it that a new version is available. You might also be able to: https://pre-commit.com/#pre-commit-try-repo